### PR TITLE
Allow copy notebook

### PR DESCRIPTION
**Pull request by antalakas**

### CODE REVIEW

File: openapi-v1.yaml [hunk 1]
```diff
@@ -5287,6 +5320,80 @@ paths:
...
+        required: true
+      - name: X-TILEDB-CLOUD-ACCESS-CREDENTIALS-NAME
+        in: header
+        description: Optional registered access credentials to use for creation
+        type: string
+        required: false
+    post:
+      description: Copy a tiledb notebook at the specified location
+      tags:
+        - files
```
Comment (SarantopoulosKon) [inline openapi-v1.yaml:5344]: Should this be in `files` package or `notebooks` ?

File: openapi-v1.yaml [hunk 2]
```diff
@@ -5287,6 +5320,80 @@ paths:
           schema:
             $ref: "#/definitions/Error"
 
+  /notebooks/{namespace}/{array}/copy:
+    parameters:
+      - type: string
```
Comment (SarantopoulosKon) [inline openapi-v1.yaml:5325]: `{array}` path param needs to be added

**Review by SarantopoulosKon — CHANGES_REQUESTED** [review]

File: openapi-v1.yaml [hunk 3]
```diff
@@ -2729,6 +2740,28 @@ definitions:
         type: string
         description: output location of the exported file
 
+  NotebookCopy:
```
Comment (anastasop) [inline openapi-v1.yaml:2743]: @antalakas some questions on the design? 
- what is output_uri? Can it be a general uri? Why we don't just copy to `default_storage_location/notebooks/new_name`
- why namespace is missing? An admin may want to copy a notebook for an organization

In general what are the drawbacks of just using `namespace, name`?

File: openapi-v1.yaml [hunk 4]
```diff
@@ -2729,6 +2740,28 @@ definitions:
...
+    type: object
+    properties:
+      output_uri:
+        type: string
+        description: output location of the TileDB File
+      name:
+        type: string
+        description: name to set for registered notebook
+
+  NotebookCopied:
```
Comment (anastasop) [inline openapi-v1.yaml:2754]: Again why don't we return `namespace, name` or a tiledb cloud uri?

File: openapi-v1.yaml [hunk 5]
```diff
@@ -5287,6 +5320,80 @@ paths:
           schema:
             $ref: "#/definitions/Error"
 
+  /notebooks/{namespace}/{array}/copy:
```
Comment (anastasop) [inline openapi-v1.yaml:5323]: The draft PR for the rest server uses timestamps for versions. We should include them in the spec

File: openapi-v1.yaml (hunk 1 above)
Comment (antalakas) [reply openapi-v1.yaml]: `notebook` thanks

File: openapi-v1.yaml (hunk 2 above)
Comment (antalakas) [reply openapi-v1.yaml:5331]: added

File: openapi-v1.yaml (hunk 3 above)
Comment (antalakas) [reply openapi-v1.yaml:2743]: `namespace` is added, let's stick with absolute `output_uri` and make the discussion regarding abstractions in later PRs when reviewing onboarding and user experience enhancements

File: openapi-v1.yaml (hunk 4 above)
Comment (antalakas) [reply openapi-v1.yaml:2757]: `name`, `namespace` added

File: openapi-v1.yaml (hunk 5 above)
Comment (antalakas) [reply openapi-v1.yaml:5329]: `end_timestamp` added

File: openapi-v1.yaml (hunk 2 above)
Comment (SarantopoulosKon) [reply openapi-v1.yaml:5331]: We need to add

```
- type: string
  name: array
  in: path
  required: true
  description: The name of the notebook
````

Trying to generate the client returns an error

```
The swagger spec at "./TileDB-Cloud-API-Spec/openapi-v1.yaml" is invalid against swagger specification 2.0. see errors :
- path param "{array}" has no parameter definition
```

**Review by SarantopoulosKon — CHANGES_REQUESTED** [review]

**Review by SarantopoulosKon — APPROVED** [review]

**Pull request merged**